### PR TITLE
self commands, including compile

### PR DESCRIPTION
**Pull request by feinstruktur**
I noticed that the vapor-cli command was getting quite slow when working on the `docker` command so I added a `bootstrap` command that will compile and install a binary `vapor` at the given path (or $CWD).

This should probably also be called automatically by `self-update`.

### CODE REVIEW

File: vapor.swift [hunk 1]
```diff
@@ -386,6 +386,41 @@ extension New {
...
+                return "./"
+            }
+        }() + "vapor"
+
+        let src = Process.arguments[0]
+        let cmd = "env SDKROOT=$(xcrun -show-sdk-path -sdk macosx) swiftc \(src) -o \(binary)"
+        do {
+            try run(cmd)
+        } catch {
+            fail("Could not compile \(src), try running the folloing command in order to debug this issue:\n\(cmd)")
```
Comment (tannernelson) [inline vapor.swift]: typo in `following`

File: vapor.swift [hunk 2]
```diff
@@ -386,6 +386,41 @@ extension New {
...
+        }
+    }
+
+    static var help: [String] {
+        return [
+                   "bootstrap <directory>",
```
Comment (tannernelson) [inline vapor.swift]: This should only be one tab indented.

**Comment by tannernelson** [comment]
This is a cool idea, I'm interested to see if it works on Linux as well.

File: vapor.swift (hunk 2 above)
Comment (feinstruktur) [inline vapor.swift]: Yeah, looks a bit out of place - this is what I get after Re-Indent. I’ll move it over and abstain from mashing the TAB key when near those lines ;)

**Comment by feinstruktur** [comment]
I’ll see if this can work on Linux as well. Don’t see why not, it’s just the `xrun` bit that needs translating.

**Comment by feinstruktur** [comment]
Halleluja

**Comment by feinstruktur** [comment]
`self-compile` may be a better name for this command, what do people think?

**Comment by tannernelson** [comment]
Yeah, let's do `self-compile`. I'll test it out and make that change now.

**Comment by feinstruktur** [comment]
I had a thought how we could solve the install step. What if on every launch the script checks (via `file`) if it's already compiled and if it isn't it self-compiles and replaces itself. It's a bit automagic but at least you wouldn't have to change the installation instructions. Should also work with self-update, which then just pulls in a fresh uncompiled version and could actually also run a compile step then.

**Pull request merged**